lib/requests.ts

```diff
@@ -13,55 +14,34 @@ export async function downloadUrl(url: string, file: string) {
 
   fs.mkdirpSync(path.dirname(tempFile));
 
-  return new Promise<request.Response>((resolve, reject) => {
-    const headers = { Accept: 'application/octet-stream' };
-    const ws = fs.createWriteStream(tempFile);
-    let result: request.Response;
+  const ws = fs.createWriteStream(tempFile);
 
-    const req = progress(
-      request.defaults({ timeout: 30 * 1000 }).get(
-        url,
-        {
-          headers,
-        },
-        (error, response) => {
-          if (error) {
-            log.disableProgress();
-            return reject(wasReported(error.message));
-          }
-          if (response.statusCode !== 200) {
-            log.disableProgress();
-            const message = `${response.statusCode} ${response.body}`;
-            return reject(wasReported(message, url));
-          }
-          result = response;
-        }
-      )
-    );
-
-    req.on('progress', (state) => {
-      let p;
+  return axios
+    .get(url, { responseType: 'stream' })
+    .then(({ data, headers }) => {
+      const totalSize = headers['content-length'];
+      let currentSize = 0;
 
-      if (state.size && state.size.transferred && state.size.total) {
-        p = state.size.transferred / state.size.total;
+      data.on('data', (chunk: string) => {
+        if (totalSize != null && totalSize !== 0) {
+          currentSize += chunk.length;
+          log.showProgress((currentSize / totalSize) * 100);
+        }
+      });
+      data.pipe(ws);
+
+      return promisify(stream.finished)(ws).then(() => {
```

Comment:
Not 100% sure but maybe this requires that the returned value is binded to stream? 
```suggestion
      return promisify(stream.finished).call(stream, ws).then(() => {
```

Reply:
Not really. I tested locally. You can use a hardcoded URL in https://github.com/vercel/pkg-fetch/blob/ba011e447459891e1c7841c4a600aa8488098364/lib/index.ts#L25. For instance, `https://github.com/jesec/pkg-fetch/releases/download/v2.7/uploaded-v2.7-node-v14.16.0-linux-x64`. 

to test yourself.